python/cudf_polars/cudf_polars/dsl/ir.py

```diff
@@ -763,11 +757,7 @@ def do_evaluate(
             c.obj.type() == dtype
             for c, dtype in zip(df.columns, schema.values(), strict=True)
         )
-        if predicate is not None:
-            (mask,) = broadcast(predicate.evaluate(df), target_length=df.num_rows)
-            return df.filter(mask)
-        else:
-            return df
+        return df
```

Comment:
Can we remove the `predicate` completely from this node (and not even both translating...)?